### PR TITLE
Fix/delete vertices

### LINKED ISSUE #330: Delete vertices => delete elements whithout notice

### PR DESCRIPTION
**Pull request by BotellaA**
fix #330

### CODE REVIEW

**Review by panquez — APPROVED** [review]
Two small modifications then OK

File: tests/model/test-relationships.cpp [hunk 1]
```diff
@@ -34,60 +34,72 @@
...
+        component_id( uuids[1] ), component_id( uuids[3] ) );
+    builder.add_boundary_relation(
+        component_id( uuids[2] ), component_id( uuids[3] ) );
+    builder.add_boundary_relation(
+        component_id( uuids[4] ), component_id( uuids[0] ) );
+    builder.add_boundary_relation(
+        component_id( uuids[4] ), component_id( uuids[5] ) );
+    builder.add_boundary_relation(
+        component_id( uuids[5] ), component_id( uuids[0] ) );
     // Repete last relation to test duplications
```
Comment (MelchiorSchuh) [inline tests/model/test-relationships.cpp:64]: Repeat

File: tests/model/test-relationships.cpp [hunk 2]
```diff
@@ -34,60 +34,72 @@
...
-    builder.add_internal_relation( uuids[4], uuids[0] );
+    builder.add_internal_relation(
+        component_id( uuids[0] ), component_id( uuids[1] ) );
+    builder.add_internal_relation(
+        component_id( uuids[2] ), component_id( uuids[3] ) );
+    builder.add_internal_relation(
+        component_id( uuids[4] ), component_id( uuids[5] ) );
+    builder.add_internal_relation(
+        component_id( uuids[4] ), component_id( uuids[0] ) );
     // Repete last relation to test duplications
```
Comment (MelchiorSchuh) [inline tests/model/test-relationships.cpp:81]: idem

File: tests/model/test-relationships.cpp [hunk 3]
```diff
@@ -34,60 +34,72 @@
...
+        component_id( uuids[1] ), component_id( uuids[5] ) );
+    builder.add_item_in_collection(
+        component_id( uuids[2] ), component_id( uuids[5] ) );
+    builder.add_item_in_collection(
+        component_id( uuids[3] ), component_id( uuids[5] ) );
+    builder.add_item_in_collection(
+        component_id( uuids[5] ), component_id( uuids[4] ) );
+    builder.add_item_in_collection(
+        component_id( uuids[0] ), component_id( uuids[4] ) );
     // Repete last relation to test duplications
```
Comment (MelchiorSchuh) [inline tests/model/test-relationships.cpp:100]: idem

**Review by MelchiorSchuh — APPROVED** [review]
Small comments

File: src/geode/mesh/builder/solid_mesh_builder.cpp [hunk 4]
```diff
@@ -191,26 +191,22 @@ namespace
...
                 const auto new_vertex =
-                    vertices_old2new[solid.polyhedron_vertex( id )];
-                if( new_vertex == geode::NO_ID )
-                {
-                    polyhedra_to_delete[p] = true;
-                }
+                    vertices_old2new[solid.polyhedron_vertex( { p, v } )];
+                OPENGEODE_EXCEPTION( new_vertex != geode::NO_ID,
+                    "[SolidMesh::update_polyhedron_vertices] No polyhedron "
+                    "should be removed" );
```
[the remark this answers is not in the document]
Comment (BotellaA) [reply src/geode/mesh/builder/solid_mesh_builder.cpp:207]: definitly exception for broken contract :p

**Pull request merged**